src/chains/ethereum/transaction/src/eip2930-access-list-transaction.ts

```diff
@@ -0,0 +1,228 @@
+import {
+  Data,
+  Quantity,
+  keccak,
+  BUFFER_EMPTY,
+  BUFFER_32_ZERO,
+  RPCQUANTITY_EMPTY
+} from "@ganache/utils";
+import { Address } from "@ganache/ethereum-address";
+import type Common from "@ethereumjs/common";
+import { BN } from "ethereumjs-util";
+import { ecsign } from "ethereumjs-util";
+import { TypedRpcTransaction } from "./rpc-transaction";
+import { encodeRange, digest } from "@ganache/rlp";
+import { RuntimeTransaction } from "./runtime-transaction";
+import {
+  EIP2930AccessListDatabasePayload,
+  EIP2930AccessListDatabaseTx,
+  TypedDatabaseTransaction
+} from "./raw";
+import { AccessList, AccessListBuffer } from "@ethereumjs/tx";
+import { AccessLists } from "./access-lists";
+import { computeInstrinsicsAccessListTx } from "./signing";
+
+const CAPABILITIES: any[] = [2718, 2930];
+export class EIP2930AccessListTransaction extends RuntimeTransaction {
+  public chainId: Quantity;
+  public accessList: AccessListBuffer;
+  public accessListJSON: AccessList;
+  public gasPrice: Quantity;
+  public type: Quantity = Quantity.from("0x1");
+
+  public constructor(
+    data: EIP2930AccessListDatabasePayload | TypedRpcTransaction,
+    common: Common
+  ) {
+    super(data, common);
+    if (Array.isArray(data)) {
+      this.chainId = Quantity.from(data[0]);
+      this.nonce = Quantity.from(data[1], true);
+      this.gasPrice = Quantity.from(data[2]);
+      this.gas = Quantity.from(data[3]);
+      this.to = data[4].length == 0 ? RPCQUANTITY_EMPTY : Address.from(data[4]);
+      this.value = Quantity.from(data[5]);
+      this.data = Data.from(data[6]);
+      const accessListData = AccessLists.getAccessListData(data[7]);
+      this.accessList = accessListData.accessList;
+      this.accessListJSON = accessListData.AccessListJSON;
+      this.v = Quantity.from(data[8]);
+      this.r = Quantity.from(data[9]);
+      this.s = Quantity.from(data[10]);
+      this.raw = [this.type.toBuffer(), ...data];
+
+      const {
+        from,
+        serialized,
+        hash,
+        encodedData,
+        encodedSignature
+      } = this.computeIntrinsics(this.v, this.raw, this.common.chainId());
+
+      this.from = from;
+      this.serialized = serialized;
+      this.hash = hash;
+      this.encodedData = encodedData;
+      this.encodedSignature = encodedSignature;
+    } else {
+      this.chainId = Quantity.from(data.chainId);
+      this.gasPrice = Quantity.from(data.gasPrice);
+      const accessListData = AccessLists.getAccessListData(data.accessList);
+      this.accessList = accessListData.accessList;
+      this.accessListJSON = accessListData.AccessListJSON;
+    }
+  }
+
+  public toJSON = () => {
+    return {
```

Comment:
The LegacyTransaction returns the type here, don't we need to? Test to make sure then add it.